### DISCUSSION TITLE
problem with gmm

### DISCUSSION THREAD

**Discussion by LaureMargarit** [stocnet/rsiena#32] [Q&A] [answered]
Hello,

I am currently doing a master thesis and I am using RSiena for my analysis, it is the first time I am using this model. I am working on networks of trade agreements. I differentiate bilateral agreements from plurilateral agreements, so I have a one-mode  and two-mode analysis.  My analysis includes 6 waves (1990, 1995, 2000, 2005, 2010, 2015).

I currently have a problem with the use of the generalized method of moments. When I run the following line: 

**AlgoEnv1 <- sienaAlgorithmCreate(projname = "Environment_in_agreement", gmm=TRUE, modelType = c(environment0=3, environment0299=3, environment35=3, environment10=3, environment10299=3, environment135=3))**

This error message appears: 
**Error in sienaAlgorithmCreate(projname = "Environment_in_agreement", : unused argument (gmm = TRUE)**

Before asking for your help I checked if I was using the right version of RSiena. So, currently I have the following version 1.2.33. I also included effects that use gmm with the function : includeGMoMStatistics().

So, I'm asking for help because I really don't understand why I have this error message. 
Thank you in advance for your answer,

Sincerely,

Laure Margarit student in Master 2 of Environment, Energy and Transport Economics at the University of Nantes

**Comment by VAmati** [comment]
Dear Laure,

I tried to reproduce the error from the information you provided in the text (i.e., using a data set studying the evolution of one-mode and/or two-mode networks) but I did not manage to obtain the same error. 
Could you please confirm that when you run the script GMOMscript.R (https://www.stats.ox.ac.uk/~snijders/siena/GMoMscript.R) everything works fine?

Thanks,
Viv

**Comment by LaureMargarit** [comment]
Dear Viv,

Thank you very much for your attention !
I run the script and had the attached error message

[image: image.png]
It seems that sienaAlgorithmCreate does not expect gmm as an argument. For
information I use RSiena version 1.3.0 and my version of RStudio is 4.0.2.

Thanks a lot for your help,

Laure

**Reply by VAmati** [reply]
Laure, I cannot see the image you attached. Could you please paste the  text of the error in the message?
Thanks,
Viv

**Comment by LaureMargarit** [comment, marked as answer]
I'm really sorry !

The mistake I had was the same:

*Error in sienaAlgorithmCreate(nsub = 2, n3 = 10000, gmm = TRUE, seed =
602) : unused argument (gmm = TRUE)*

I have just installed the new version of R 4.1.0 and the error does not
appear anymore...

Thank you so much for taking the time to look!

Laure

**Reply by VAmati** [reply]
Laure, thank you for letting me know that the problem is solved!
Best,
Viv
